### ISSUE TITLE
Linting doesn't work over plpgsql blocks

### ISSUE THREAD

**Issue by galhar** [sqlfluff/sqlfluff#5864]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I really want to use sqlfluff in my system contains multiple, big sql files runned over greenplum.
The problem is that many of my sql files contains plpgsql blocks, and sqlfluff seems to not work when being run over such files. 

### Expected Behaviour

running lint should yeild errors, running fix should fix the file, when running over the following code of plpgsql block:
```sql
do
$$
declare 
  v1 INTEGER;
begin
  create temp table bla3 as
  with bla1 as (select generate_series(0,100,1) as x1
  ),
  bla2 as (
        select x1, x1*2 as x2
                 from bla1
  )
  select 
    bla1.x1, bla2.x2,
         bla2.x2 * bla1.x1 as x3
  from bla2
inner join bla1
    on true;
  analyze bla3;
  
  raise info 'max is %', (select max(x3) from bla3);  
 
end;
$$
language plpgsql;
```

the same way it would work over this same code out of the plpgsql block:
```sql
create temp table bla3 as
  with bla1 as (select generate_series(0,100,1) as x1
  ),
  bla2 as (
        select x1, x1*2 as x2
                 from bla1
  )
  select 
    bla1.x1, bla2.x2,
         bla2.x2 * bla1.x1 as x3
  from bla2
inner join bla1
    on true;
  analyze bla3;
```

### Observed Behaviour

But I get the following results when fixing:
The plpgsql remainx exactly the same:
```sql
do
$$
declare 
  v1 INTEGER;
begin
  create temp table bla3 as
  with bla1 as (select generate_series(0,100,1) as x1
  ),
  bla2 as (
        select x1, x1*2 as x2
                 from bla1
  )
  select 
    bla1.x1, bla2.x2,
         bla2.x2 * bla1.x1 as x3
  from bla2
inner join bla1
    on true;
  analyze bla3;
  
  raise info 'max is %', (select max(x3) from bla3);  
 
end;
$$
language plpgsql;
```

whereas the same code not within such a block get fixed as desired:
```sql
create temp table bla3 as
with bla1 as (select generate_series(0, 100, 1) as x1
),

bla2 as (
    select
        x1,
        x1 * 2 as x2
    from bla1
)

select
    bla1.x1,
    bla2.x2,
    bla2.x2 * bla1.x1 as x3
from bla2
inner join bla1
    on true;
analyze bla3;
```

### How to reproduce

Just run `sqlfluff fix example.sql --dialect greenplum`, when the example.sql file contains either one of the code blocks wrote above.

### Dialect

greenplum

### Version

3.0.6

### Configuration

default, no configuration file needed

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by galhar** [comment]
I might be wrongly labelling it as "bug", if it isn't supported by purpose. But it seems like Sqlfluff should support postgresql and greenplum.

**Comment by corpix** [comment]
It is even doesn't format SQL in this blocks which is kinda... unexpected
